### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by zlayabekrija**
Review pleeeeeease 😄

### CODE REVIEW

**Comment by heyapricot** [comment]
You get coolness points from your repository's name 😎

**Comment by heyapricot** [comment]
## Project Requirements
### General

- [ ] Project has a descriptive README.md

### `Ship`

- [ ] There is a `Ship` `Factory function`
- [ ] has a `length` property
- [ ] can tell if it has been sunk
- [ ] Has a hit method that takes a number as an argument and marks that position as hit

### `Gameboard`

- [ ] Has a `Factory Function`
- [ ] Can instantiate a `Ship` at specific coordinates
- [ ] Keeps track of missed attacks
- [ ] Can tell if all it's `Ship` have been sunk

#### `receiveAttack`

- [ ]  takes a pair of coordinates as argument. 
- [ ] determines if the attack hits a ship
- [ ] calls the `Ship` `hit` method

### `Player`

- [ ] Alternates between turns

## General

- [ ] You can play against a programmed player that makes random plays
- [ ] It has an interface
- [ ] It displays two boards (One for each player)
- [ ] Allows to click a coordinate in an enemy board
- [ ] The game loop uses methods from other objects

**Comment by heyapricot** [comment]
# Color me in 🌈
I like the way you designed the interface and the brief flash when hovering over a cell. It could use a bit more color though. [I like to use color palettes from colourlovers](https://www.colourlovers.com/)

File: src/js/Ship.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const Ship = (name, ship) => {
+  let size = new Array(ship).fill(name);
+  const hit = (position, size) => size[position] = 'X';
+  const isSunk = (size) => size.every(pos);
+  const pos = (pos) => pos === 'X';
```
[the remark this answers is not in the document]
Comment (zlayabekrija) [reply src/js/Ship.js]: Converted :)

File: src/js/Gameboard.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  }
+  const position = (direction, ship, startingX, startingY) => {
+    return direction === 'h' ? horizontal(ship, startingX, startingY) : vertical(ship, startingX, startingY)
+  }
+  const invalidH = (x, y, arr, compare) => {
+    for (let i = 0; i < compare.length; i++) {
+      if (arr[y][x + i] !== '0') return true;
+    }
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (zlayabekrija) [reply src/js/Gameboard.js]: Personally I prefer this way

**Comment by zlayabekrija** [comment]
Thanks a lot, your guidelines were very helpful ❤️

**Pull request merged**